### PR TITLE
Add JSDoc for bundles

### PR DESCRIPTION
**Pull request by wu-hui**

### CODE REVIEW

File: packages/firebase/index.d.ts [hunk 1]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
     INTERNAL: { delete: () => Promise<void> };
   }
 
+  /**
+   * Represents the task of loading a Firestore bundle. It provides progress of the bundle
+   * loading, task completion and error events should they be any.
+   *
+   * It can be used as a `Promise<LoadBundleTaskProgress>`.
```
Comment (schmidt-sebastian) [inline packages/firebase/index.d.ts:8326]:
```suggestion
   * The API is compatible with `Promise<LoadBundleTaskProgress>`.
```

File: packages/firebase/index.d.ts [hunk 2]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+     * @param complete Called when the loading task is complete.
+     */
     onProgress(
       next?: (progress: LoadBundleTaskProgress) => any,
       error?: (error: Error) => any,
       complete?: () => void
     ): void;
 
+    /**
+     * Implements a `Promise<LoadBundleTaskProgress>` interface.
```
Comment (schmidt-sebastian) [inline packages/firebase/index.d.ts:8344]:
```suggestion
     * Implements the `Promise<LoadBundleTaskProgress>.then` interface.
```

File: packages/firebase/index.d.ts [hunk 3]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+     *        loading task completes.
+     * @param onRejected It is called when there is an error occurred from loading the bundle.
+     */
     then<T, R>(
       onFulfilled?: (a: LoadBundleTaskProgress) => T | PromiseLike<T>,
       onRejected?: (a: Error) => R | PromiseLike<R>
     ): Promise<T | R>;
 
+    /**
+     * Implements a `Promise<LoadBundleTaskProgress>` interface.
```
Comment (schmidt-sebastian) [inline packages/firebase/index.d.ts:8356]:
```suggestion
     * Implements the `Promise<LoadBundleTaskProgress>.catch` interface.
```

File: packages/firebase/index.d.ts [hunk 4]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
     INTERNAL: { delete: () => Promise<void> };
   }
 
+  /**
+   * Represents the task of loading a Firestore bundle. It provides progress of the bundle
+   * loading, task completion and error events should they be any.
+   *
+   * It can be used as a `Promise<LoadBundleTaskProgress>`.
+   */
   export interface LoadBundleTask {
```
Comment (schmidt-sebastian) [inline packages/firebase/index.d.ts:8328]:
```suggestion
  export interface LoadBundleTask extends PromiseLIke< LoadBundleTaskProgress> {
```

File: packages/firebase/index.d.ts [hunk 5]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
     catch<R>(
       onRejected: (a: Error) => R | PromiseLike<R>
     ): Promise<R | LoadBundleTaskProgress>;
   }
 
+  /**
+   * Represents a progress update or a final state from loading bundles.
+   */
   export interface LoadBundleTaskProgress {
+    /** How many documents have been loaded. */
```
Comment (schmidt-sebastian) [inline packages/firebase/index.d.ts:8369]: I noticed that you removed these in the other PR. You should leave the comments in both places as firestore-exp builds its d.ts file from source.

**Comment by markarndt** [comment]
Few edits sprinkled in above....

File: packages/firebase/index.d.ts (hunk 5 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts:8369]: Yeah, that was a bad merge i think. Will keep the comments.

File: packages/firebase/index.d.ts [hunk 6]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+   * Represents the task of loading a Firestore bundle. It provides progress of the bundle
+   * loading, task completion and error events should they be any.
+   *
+   * The API is compatible with `Promise<LoadBundleTaskProgress>`.
+   */
+  export interface LoadBundleTask extends PromiseLike<LoadBundleTaskProgress> {
+    /**
+     * Registers functions to listen to bundle loading progresses.
+     * @param next Called there is a progress update from bundle loading, typically whenever
+     *        a Firestore document is loaded it will generate a progress update.
```
Comment (schmidt-sebastian) [inline packages/firebase/index.d.ts:8332]: Please fix indent to match the rest of this file.

**Review by schmidt-sebastian — APPROVED** [review]

File: packages/firebase/index.d.ts (hunk 6 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts]: Done.

File: packages/firebase/index.d.ts [hunk 7]
```diff
@@ -8291,10 +8291,26 @@ declare namespace firebase.firestore {
...
+     */
     loadBundle(
-      bundleData: ArrayBuffer | ReadableStream<ArrayBuffer> | string
+      bundleData: ArrayBuffer | ReadableStream<Uint8Array> | string
     ): LoadBundleTask;
 
+    /**
+     * Reads a Firestore query from local cache that is associated to a given name.
+     *
+     * The named queries are from bundles, and saved as a result of `loadBundle`. `namedQuery`
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts:8312]: Done..more or less similar.

File: packages/firebase/index.d.ts [hunk 8]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
     INTERNAL: { delete: () => Promise<void> };
   }
 
+  /**
+   * Represents the task of loading a Firestore bundle. It provides progress of the bundle
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts:8325]: Done.

File: packages/firebase/index.d.ts [hunk 9]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+   * loading, task completion and error events should they be any.
+   *
+   * It can be used as a `Promise<LoadBundleTaskProgress>`.
+   */
   export interface LoadBundleTask {
+    /**
+     * Registers functions to listen to bundle loading progresses.
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts:8332]: Done.

File: packages/firebase/index.d.ts [hunk 10]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+     * @param next Called there is a progress update from bundle loading, typically whenever
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts]: Done.

File: packages/firebase/index.d.ts [hunk 11]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+     *        a Firestore document is loaded it will generate a progress update.
+     * @param error Called when there is an error occurred from loading the bundle. The task
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts]: Done. Not sure what you mean by second statement.

File: packages/firebase/index.d.ts [hunk 12]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
     onProgress(
       next?: (progress: LoadBundleTaskProgress) => any,
       error?: (error: Error) => any,
       complete?: () => void
     ): void;
 
+    /**
+     * Implements a `Promise<LoadBundleTaskProgress>` interface.
+     *
+     * @param onFulfilled It is called with the compeltion `LoadBundleTaskProgress` when the
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts]: There is no status code.

File: packages/firebase/index.d.ts [hunk 13]
```diff
@@ -8303,31 +8319,71 @@ declare namespace firebase.firestore {
...
+     */
     then<T, R>(
       onFulfilled?: (a: LoadBundleTaskProgress) => T | PromiseLike<T>,
       onRejected?: (a: Error) => R | PromiseLike<R>
     ): Promise<T | R>;
 
+    /**
+     * Implements a `Promise<LoadBundleTaskProgress>` interface.
+     *
+     * @param onRejected It is called when there is an error occurred from loading the bundle.
```
[the remark this answers is not in the document]
Comment (wu-hui) [reply packages/firebase/index.d.ts]: Done, but without mentioning status code.

File: packages/firebase/index.d.ts [hunk 14]
```diff
@@ -8291,10 +8289,28 @@ declare namespace firebase.firestore {
      */
     terminate(): Promise<void>;
 
+    /**
+     * Loads a Firestore bundle into the local cache.
+     *
+     * @param bundleData
+     *   An object representing the bundle to be load. Valid objects are `ArrayBuffer`,
```
Comment (markarndt) [inline packages/firebase/index.d.ts:8296]: "...bundle to be loaded. Valid..."

File: packages/firebase/index.d.ts [hunk 15]
```diff
@@ -8291,10 +8289,28 @@ declare namespace firebase.firestore {
...
+     *   `ReadableStream<Uint8Array>` or `string`.
+     *
+     * @return
+     *   A `LoadBundleTask` object, which notifies callers with progress update, and completion
```
Comment (markarndt) [inline packages/firebase/index.d.ts:8300]: "...with progress updates, and..."

File: packages/firebase/index.d.ts [hunk 16]
```diff
@@ -8303,31 +8319,77 @@ declare namespace firebase.firestore {
...
+   * Represents the task of loading a Firestore bundle. It provides progress of bundle
+   * loading, as well as task completion and error events.
+   *
+   * The API is compatible with `Promise<LoadBundleTaskProgress>`.
+   */
+  export interface LoadBundleTask extends PromiseLike<LoadBundleTaskProgress> {
+    /**
+     * Registers functions to listen to bundle loading progress events.
+     * @param next
+     *   Called when there is a progress update from bundle loading, typically `next` calls occur
```
Comment (markarndt) [inline packages/firebase/index.d.ts:8332]: ...from bundle loading. Typically, `next` calls..."

File: packages/firebase/index.d.ts [hunk 17]
```diff
@@ -8303,31 +8319,77 @@ declare namespace firebase.firestore {
...
       next?: (progress: LoadBundleTaskProgress) => any,
       error?: (error: Error) => any,
       complete?: () => void
     ): void;
 
+    /**
+     * Implements the `Promise<LoadBundleTaskProgress>.then` interface.
+     *
+     * @param onFulfilled
+     *   Called on the completion with a `LoadBundleTaskProgress` update when the
```
Comment (markarndt) [inline packages/firebase/index.d.ts:8350]: This one was a little unclear. Maybe this edit will help, but please clarify "the completion of" what? Completion of the progress update?

"Called on completion of a `LoadBundleTaskProgress` update when..."

File: packages/firebase/index.d.ts [hunk 18]
```diff
@@ -8291,10 +8289,28 @@ declare namespace firebase.firestore {
...
+     *   A `LoadBundleTask` object, which notifies callers with progress update, and completion
+     *   or error events. It can be used as a `Promise<LoadBundleTaskProgress>`.
+     */
     loadBundle(
-      bundleData: ArrayBuffer | ReadableStream<ArrayBuffer> | string
+      bundleData: ArrayBuffer | ReadableStream<Uint8Array> | string
     ): LoadBundleTask;
 
+    /**
+     * Reads a Firestore `Query` from local cache that is associated to a given name.
```
Comment (markarndt) [inline packages/firebase/index.d.ts:8308]: "Reads a Firestore `Query` from local cache, identified by name."

"Bundles can contain named queries. These named queries are packaged..."

**Review by markarndt — APPROVED** [review]

File: packages/firebase/index.d.ts (hunk 14 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts:8296]: Done.

File: packages/firebase/index.d.ts (hunk 15 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts:8300]: Done.

File: packages/firebase/index.d.ts (hunk 18 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts:8308]: Done.

File: packages/firebase/index.d.ts (hunk 16 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts:8332]: Done.

File: packages/firebase/index.d.ts (hunk 17 above)
Comment (wu-hui) [reply packages/firebase/index.d.ts:8350]: Done.

**Review by markarndt — APPROVED** [review]
Great.

**Pull request merged**